tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['key_format'] == 'PASSPHRASE', str(job_status_result)
+            assert dictionary['unlock_successful'] is True, str(job_status_result)
+            assert dictionary['locked'] is False, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_22_delete_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_23_delete_pool():
+    payload = {
+        'cascade': True,
+        'restart_services': True,
+        'destroy': True
+    }
+    results = POST(f'/pool/id/{pool_id}/export/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_24_create_a_passphrase_encrypted_pool():
+    global pool_id
+    payload = {
+        'name': pool_name,
+        'encryption': True,
+        'encryption_options': {
+            'algorithm': 'AES-128-CCM',
+            'passphrase': 'my_pool_passphrase',
+        },
+        'topology': {
+            'data': [
+                {'type': 'STRIPE', 'disks': disk_pool}
+            ],
+        }
+    }
+    results = POST('/pool/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    pool_id = job_status['results']['result']['id']
+
+
+@skip_for_ha
+def test_25_create_a_passphrase_encrypted_root_on_passphrase_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'generate_key': False,
+            'pbkdf2iters': 100000,
+            'algorithm': 'AES-128-CCM',
+            'passphrase': 'my_passphrase',
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_26_change_a_passphrase_encrypted_root_to_key_on_passphrase_encrypted_pool():
+    payload = {
+        'id': dataset,
+        'change_key_options': {
+            'key': dataset_token_hex,
+        }
+    }
+    results = POST(f'/pool/dataset/change_key/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
```

Comment:
This is going to fail ( please use latest nightlies to test this ), and we should mark it as such. It is not allowed for a passphrase encrypted root to have children encrypted roots which are encrypted with a `KEY`.